obspy/io/nordic/core.py

```diff
@@ -50,51 +57,74 @@ def __init__(self, value):
         self.value = value
 
 
-def _is_sfile(sfile):
+def _is_sfile(sfile, encoding='latin-1'):
     """
     Basic test of whether the file is nordic format or not.
 
     Not exhaustive, but checks some of the basics.
 
     :type sfile: str
     :param sfile: Path to sfile
+    :type encoding: str
+    :param encoding: Encoding for file, used to decode from bytes to string
+
     :rtype: bool
     """
     if not hasattr(sfile, "readline"):
         try:
-            with open(sfile, 'r') as f:
-                head_line = _get_headline(f)
+            with open(sfile, 'r', encoding=encoding) as f:
+                tags = _get_line_tags(f=f)
         except Exception:
             return False
     else:
-        head_line = _get_headline(sfile)
-    if head_line is not None:
+        tags = _get_line_tags(f=sfile)
+    if tags is not None:
+        # Note that there can be two origin lines, but only to allow more
+        # magnitudes, not more origins.
+        try:
+            head_line = tags['1'][0][0]
+        except KeyError:
+            return False
         try:
             sfile_seconds = int(head_line[16:18])
         except ValueError:
             return False
         if sfile_seconds == 60:
             sfile_seconds = 0
         try:
-            UTCDateTime(int(head_line[1:5]), int(head_line[6:8]),
-                        int(head_line[8:10]), int(head_line[11:13]),
-                        int(head_line[13:15]), sfile_seconds,
-                        int(head_line[19:20]) * 100000)
+            UTCDateTime(
+                int(head_line[1:5]), int(head_line[6:8]), int(head_line[8:10]),
+                int(head_line[11:13]), int(head_line[13:15]), sfile_seconds,
+                int(head_line[19:20]) * 100000)
             return True
         except Exception:
             return False
     else:
         return False
 
 
-def _get_headline(f):
+def _get_line_tags(f):
+    f.seek(0)
+    line = f.readline()
+    if len(line.rstrip('\n').rstrip('\r')) != 80:
+        # Cannot be Nordic
+        raise NordicParsingError(
+            "Lines are not 80 characters long: not a nordic file")
+    f.seek(0)
+    tags = {}
     for i, line in enumerate(f):
-        if i == 0 and len(line.rstrip()) != 80:
-            return None
-        if line[79] == '1':
-            return line
-    else:
-        return None
+        try:
+            line_id = line[79]
+        except IndexError:
+            line_id = ' '
+        if line_id in tags.keys():
+            tags[line_id].append((line, i))
+        elif line_id in accepted_tags:
+            tags.update({line_id: [(line, i)]})
```

Comment:
This would be about the same number of lines but it feels a bit more elegant to me:

```python
tags = collections.defaultdict(list)
...
if line_id in accepted_tags:
    tags[line_id].append((line, i))
```